assignment1/src/main/java/ro/utcn/sd/he/assignment1/controller/CommandLineController.java

```diff
@@ -86,24 +89,160 @@ private boolean handleCommand(String command){
+        int id = scanner.nextInt();
+        scanner.nextLine();
+        print("+1 or -1?");
+        int vote = scanner.nextInt();
+        scanner.nextLine();
+        if(type.equals("question")){
+            Question question = questionService.findById(id);
+            if(vote == +1){
+                voteService.voteUp(user, question);
+            }
+            else if(vote == -1){
+                voteService.voteDown(user,question);
+            } else {
+                print("invalid vote option");
+            }
+        } else if ( type.equals("answer")){
+            Answer answer = answerService.findById(id);
+            if(vote == +1){
+                voteService.voteUp(user, answer);
+            }
+            else if(vote == -1){
+                voteService.voteDown(user,answer);
+            } else {
+                print("invalid vote option");
+            }
+        } else {
+            print("You can only vote a question or an answer");
+        }
+    }
+
+    private void handleShow(){
+        print("Select question(id):");
+        int questionId = scanner.nextInt();
+        Question question = questionService.findById(questionId);
+        if(question != null){
+            List<Answer> answers = answerService.findAnswersOf(question);
+            print(question.toString() + "Votes: " + voteService.getVoteCount(question));
+            for(Answer a : answers){
+                print("\t" + a.toString() + "Votes: " + voteService.getVoteCount(a));
+            }
+
+        } else {
+            print("the question doesn't exist");
+        }
+
+    }
+
+    private void handleAnswer(){
+        if (isLogged()){
+            print("what question(id) do you want to answer:");
+            int questionId = scanner.nextInt();
+            scanner.nextLine(); //dump line
+            print("give your answer:");
+            String text = scanner.nextLine();
+            answerService.saveAnswer(new Answer(0,user.getUsername(),text, new Timestamp(System.currentTimeMillis()),questionId));
+
+        } else {
+            print("You need to login to answer a question");
+        }
+    }
+
     private void handleSearch(){
         print("what search criteria you want to use(title/tags):");
         String criteria = scanner.next().trim();
+        scanner.nextLine();
         if(criteria.equals("title")){
             print("Enter title to search:");
-            String title1 = scanner.nextLine();///whyyyyyy
-            String title = scanner.nextLine();///whyyyyyy
+            String title = scanner.nextLine();
             List<Question> questions = questionService.listQuestions();
             questions.sort(Comparator.comparing(Question::getCreation_date_time).reversed());;
             questions.forEach( question -> {
-                if(question.getTitle().indexOf(title) > -1){
+                if(question.getTitle().toLowerCase().indexOf(title.toLowerCase()) > -1){
```

Comment:
Strings have a `contains` method that you can use instead of `indexOf(...) > -1`.